src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisWorkerService.java

```diff
@@ -223,13 +224,9 @@ public void setSecurityAnalysisFactorySupplier(Function<String, SecurityAnalysis
             if (future != null) {
                 future.cancel(true);  // cancel computation in progress
 
-                resultRepository.delete(cancelContext.getResultUuid())
-                        .doOnSuccess(unused -> {
-                            stoppedPublisherService.publishCancel(cancelContext.getResultUuid(), cancelContext.getReceiver());
-                            LOGGER.info(CANCEL_MESSAGE + " (resultUuid='{}')", cancelContext.getResultUuid());
-                        })
-                        .doOnError(throwable -> LOGGER.error(throwable.toString(), throwable))
-                        .subscribe();
+                resultRepository.delete(cancelContext.getResultUuid());
+                stoppedPublisherService.publishCancel(cancelContext.getResultUuid(), cancelContext.getReceiver());
+                LOGGER.info(CANCEL_MESSAGE + " (resultUuid='{}')", cancelContext.getResultUuid());
```

Comment:
Handling error case has disappeared ...

Reply:
Don't know against what to protect. Should I wrap in try/catch(Throwable) ?